pages/api/download-subdirectory.tsx

```diff
@@ -0,0 +1,33 @@
+import { NextApiRequest, NextApiResponse } from 'next';
+import JSZip from 'jszip';
+/**
+ * request handler for subdirectory download
+ * @param req - request body
+ * @param res - response body
+ */
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  try {
+    const data = req.body;
+    const response = await fetch(data.url);
+    const array = await response.arrayBuffer();
+
+    const zip = await JSZip.loadAsync(array, {});
+    const rootname = Object.keys(zip.files)[0];
+    const subdirZip = zip.folder(rootname)?.folder(data.subdirectory);
+
+    if (subdirZip === null || subdirZip === undefined) {
+      throw TypeError('subdirectory does not exist');
+    }
+    const base64send = await subdirZip.generateAsync({ type: 'base64' });
+
+    res.status(200);
+    res.send(base64send);
+    res.end();
+  } catch (error) {
+    res.json(error);
+    res.status(405).end();
```

Comment:
should this be 404 if subdir does not exist?

Reply:
this catches any error in the code above so it should probably be 404 if its the thrown error, else maybe 500 for errors in any other part of the code ?

Reply:
added this to issue -- resolved

Reply:
yes that is right. can be 404 when we cant find the subdir, 500 otherwise

Reply:
This still needs to be fixed I think